### PR TITLE
[curl] add url option to disable peer verification

### PR DESCRIPTION
**Pull request by Rechi**
## Motivation and Context
implements feature requested at https://trac.kodi.tv/ticket/17878 and in the forum

## How Has This Been Tested?
```<sources>
	<files>
		<source>
			<name>_expired.badssl.com</name>
			<path pathversion="1">https://expired.badssl.com/|verifypeer=false</path>
		</source>
		<source>
			<name>_wrong.host.badssl.com</name>
			<path pathversion="1">https://wrong.host.badssl.com/</path>
		</source>
		<source>
			<name>_self-signed.badssl.com</name>
			<path pathversion="1">https://self-signed.badssl.com/|verifypeer=false</path>
		</source>
		<source>
			<name>_untrusted-root.badssl.com</name>
			<path pathversion="1">https://untrusted-root.badssl.com/|verifypeer=false</path>
		</source>
		<source>
			<name>_revoked.badssl.com</name>
			<path pathversion="1">https://revoked.badssl.com/</path>
		</source>
	</files>
</sources>
```

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by saunaants** [comment]
wasnt there a decision against using | url options after the proxy leakage and header filtering debacle

**Review by wsnipex — APPROVED** [review]

**Comment by JohnPlayerSpecial2018** [comment]
@Rechi 
unfortunately not yet working for added media sources like webdavs/sftp
log: https://forum.kodi.tv/showthread.php?tid=331918&pid=2737146#pid2737146

**Comment by Rechi** [comment]
If there is an explicit warning show, when disabling peer verification, it would be fine for me.

**Comment by rbuehlma** [comment]
@Rechi I am facing an issue with this PR, to be exact, with this commit: https://github.com/xbmc/xbmc/pull/13909/commits/78d2c82d50ddcd7daf6cf3cd8e5d80ea91a458ba

Here is where I parse (in the PVR addon) the set-cookie header:
https://github.com/rbuehlma/pvr.zattoo/blob/f04367b3ae9a0aa9bb5670d033f3f419ab54944c/src/Curl.cpp#L96

After your commit, I receive the cookie value as:
 "<the expected value>\054"
The quotes are part of the value that is read from the cookie.
If I revert the above commit, I only receive
<the expected value>

I do not see how this is related to your change but it has to be as reverting it, solves the problem. It still could be that the bug is somewhere else. Do you see the issue or the relation to your change?

File: xbmc/filesystem/CurlFile.cpp [hunk 1]
```diff
@@ -837,6 +841,11 @@ void CCurlFile::ParseAndCorrectUrl(CURL &url2)
         {
           SetCustomRequest(value);
         }
+        if (name == "verifypeer")
```
Comment (Rechi) [inline xbmc/filesystem/CurlFile.cpp]: @rbuehlma try with `else if`

I'll fix it.

Comment (rbuehlma) [reply xbmc/filesystem/CurlFile.cpp]: Yap, I can confirm this does fix my issue :) Thank you very much.

**Comment by JohnPlayerSpecial2018** [comment]
@Rechi I suppose there is another problem with verifypeer=false in connection with the library. 
I posted in forum: https://forum.kodi.tv/showthread.php?tid=331918&pid=2742328#pid2742328

**Pull request merged**